lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -1418,8 +1418,32 @@ SmallVector<Direction> PrimitiveOp::portDirections() {
 
 SmallVector<DictionaryAttr> PrimitiveOp::portAttributes() {
   SmallVector<DictionaryAttr> portAttributes;
-  for (size_t i = 0; i < getReferencedPrimitive().getAllPorts().size(); ++i)
-    portAttributes.push_back(DictionaryAttr());
+  OpBuilder builder(getContext());
+  auto cleanAttrs = [&](DictionaryAttr dict){
+    if (dict) {
+      llvm::SmallVector<NamedAttribute> attrs;
+      for (NamedAttribute attr : dict) {
+        Dialect *dialect = attr.getNameDialect();
+        if (dialect != nullptr && isa<CalyxDialect>(*dialect)) {
```

Comment:
Similar here, 

if (dialect == nullptr || !isa<CalyxDialect>(*dialect)) continue;

https://releases.llvm.org/10.0.0/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code